### PR TITLE
More ec2 pipelines using dynamic PATs

### PR DESCRIPTION
**Pull request by supertopher**
parker noticed we missed a few
reviewer: please help me make sure the function exists
for `find_valid_pat` in each place it's called

## What
Use the dynamic PAT logic everywhere an ec2 runner is called

## How
Actually call and use the runner in all runner spots

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -595,9 +599,9 @@ jobs:
         id: start-ec2-runner
         uses: ./.github/actions/start-aws-runner
         with:
+          github-token: ${{ needs.find_valid_pat.outputs.pat }}
```
Comment (supertopher) [inline .github/workflows/gradle.yml:602]: this one looks weird, but it's just up two lines.  I think it's okay

File: .github/workflows/test-performance-command.yml [hunk 2]
```diff
@@ -43,6 +43,7 @@ jobs:
             ${{ secrets.DAVINCHIA_PAT }}
   start-test-runner:
     name: Start Build EC2 Runner
+    needs: find_valid_pat
```
Comment (supertopher) [inline .github/workflows/test-performance-command.yml:46]: double checked this obscure one for the function.  looks good

File: .github/workflows/release-airbyte-os.yml [hunk 3]
```diff
@@ -29,6 +29,7 @@ jobs:
   # In case of self-hosted EC2 errors, remove this block.
   start-release-airbyte-runner:
     name: "Release Airbyte: Start EC2 Runner"
+    needs: find_valid_pat
```
Comment (supertopher) [inline .github/workflows/release-airbyte-os.yml:32]: double checked this one for a function LGTM

**Review by pmossman — APPROVED** [review]
LGTM!

File: .github/workflows/publish-external-command.yml [hunk 4]
```diff
@@ -42,6 +42,7 @@ jobs:
   # In case of self-hosted EC2 errors, remove this block.
   start-publish-image-runner:
     name: Start Build EC2 Runner
+    needs: find_valid_pat
```
Comment (supertopher) [inline .github/workflows/publish-external-command.yml:45]: double checked for function

**Pull request merged**